starknet_py/net/client.py

```diff
@@ -179,10 +179,7 @@ async def wait_for_tx(
                     raise TransactionRejectedError(message=tx_receipt.rejection_reason)
 
                 if execution_status == TransactionExecutionStatus.REVERTED:
-                    # TODO (#1047): message should be always revert_reason once GatewayClient is deprecated
-                    raise TransactionRevertedError(
-                        message=(tx_receipt.revert_reason or tx_receipt.revert_error)
-                    )
+                    raise TransactionRevertedError(message=tx_receipt.revert_reason)
```

Comment:
This is not covered by codecov because of old devnet, right?

Reply:
Currently it is covered in `tests_on_networks/client_test.py` [here](https://github.com/software-mansion/starknet.py/blob/development/starknet_py/tests/e2e/tests_on_networks/client_test.py#L64), but in this PR those tests failed. It seems to be some general issue with integration network, that needs to be investigated further.

Reply:
Sure, but later we'll still need to create new tests on integration, since it is the first network to receive changes (if we want to be up-to-date with RPC updates).

Reply:
>  we'll still need to create new tests on integration, since it is the first network to receive changes

Sure, but if we merge testnet and integration tests, we could still pass integration-only data to parametrized test, until the changes are not available on testnet.

Reply:
Thanks! Added this point in #1219 to consider this during tests refactoring.